packages/store/tools/arb-passable.js

```diff
@@ -0,0 +1,89 @@
+// @ts-check
+import { Far, makeTagged } from '@endo/marshal';
+import { fc } from '@fast-check/ava';
+import '../src/types.js';
+
+/**
+ * The only elements with identity. Everything else should be equal
+ * by contents.
+ */
+export const exampleAlice = Far('alice', {});
+export const exampleBob = Far('bob', {});
+export const exampleCarol = Far('carol', {});
+
+export const arbString = fc.oneof(fc.string(), fc.fullUnicodeString());
+
+export const arbLeaf = fc.oneof(
+  fc.record({}),
+  arbString,
+  arbString.map(s => Symbol.for(s)),
+  // primordial symbols and registered lookalikes
+  fc.constantFrom(
+    ...Object.getOwnPropertyNames(Symbol).flatMap(k => {
+      const v = Symbol[k];
+      if (typeof v !== 'symbol') return [];
+      return [v, Symbol.for(k), Symbol.for(`@@${k}`)];
+    }),
+  ),
+  fc.bigInt(),
+  fc.integer(),
+  fc.constantFrom(-0, NaN, Infinity, -Infinity),
+  fc.constantFrom(null, undefined, false, true),
+  fc.constantFrom(exampleAlice, exampleBob, exampleCarol),
+  arbString.map(s => new Error(s)),
+  // unresolved promise
+  fc.constant(new Promise(() => {})),
+);
+
+const { arbDag } = fc.letrec(tie => {
+  return {
+    arbDag: fc.oneof(
+      {
+        // depthFactor: 0.5, // TODO Replace with what from new API?
+        withCrossShrink: true,
+      },
+      arbLeaf,
+      tie('arbDag').map(v => Promise.resolve(v)),
+      fc.array(tie('arbDag'), { maxLength: 3 }),
+      fc.dictionary(
+        arbString.filter(s => s !== 'then'),
+        tie('arbDag'),
+        { maxKeys: 3 },
```

Comment:
The improvements for recursive arbitraries also allow us to remove these size constraints.
```suggestion
      fc.array(tie('arbDag')),
      fc.dictionary(
        arbString.filter(s => s !== 'then'),
        tie('arbDag'),
```

Reply:
Done.